src/shared/functions.py

```diff
@@ -108,29 +112,34 @@ def update_metadata(
     if extra_items is None:
         extra_items = {}
     check_meta_type(meta_type)
+    logger.info(f"### Updating metadata {meta_type}")
+    logger.info(f"{study} {data_package} {version}")
+    logger.info(f"Key: {target} Value: {value}")
+    logger.info(f"Pre-update size: {len(metadata.keys)}")
+
     match meta_type:
         case enums.JsonFilename.TRANSACTIONS.value:
             site_metadata = metadata.setdefault(site, {})
             study_metadata = site_metadata.setdefault(study, {})
             data_package_metadata = study_metadata.setdefault(data_package, {})
             data_version_metadata = data_package_metadata.setdefault(
-                version, TRANSACTION_METADATA_TEMPLATE
+                version, copy.deepcopy(TRANSACTION_METADATA_TEMPLATE)
```

Comment:
nit: this kind of tricky nuance repeated multiple times gives me pause - maybe make a utility function like `_update_metadata` that takes some args, I dunno. Or just leave it as is.

Reply:
all three stages do something :slightly: different, but i could try to bundle this up a bit better/make it more readable